src/main/java/ir/sahab/kafkaconsumer/OffsetTracker.java

```diff
@@ -195,7 +231,7 @@ OptionalLong ack(long offset) {
             if (pageIndex <= lastConsecutivePageIndex) {
                 logThrottle.logger("redundant-page").warn("An ack received which completes a page "
                                 + "but the page is already completed. It is valid if it has "
-                                + "been a re-balance recently. offset: {}, completed page index: {}",
+                                + "been a re-balance or cleanup recently. offset: {}, completed page index: {}",
```

Comment:
```suggestion
                                + "been a re-balance recently. offset: {}, completed page index: {}",
```
According to the current implementation, we do not see a surprising 'ack' because of cleanup.